server/bundle.go

```diff
@@ -4,15 +4,25 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
+	"log"
 	"net/http"
 
 	"github.com/julienschmidt/httprouter"
 
+	"github.com/ndlib/bendo/items"
 	"github.com/ndlib/bendo/store"
 )
 
 // BundleListHandler handles GET requests to "/bundle/list".
 func (s *RESTServer) BundleListHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	if s.TapeUsed == false {
+		w.WriteHeader(503)
+		fmt.Fprintln(w, items.ErrNoStore)
```

Comment:
I guess , for me, it boils down to where error messages for a class should
be kept. In the class- encapsulation. SOP in Java.

On Thu, Aug 25, 2016 at 4:52 PM, Don Brower <notifications@github.com>
wrote:

> In server/bundle.go
> <https://github.com/ndlib/bendo/pull/136#discussion_r76322302>:
>
> >  	"github.com/ndlib/bendo/store"
> >  )
> >
> >  // BundleListHandler handles GET requests to "/bundle/list".
> >  func (s *RESTServer) BundleListHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
> > +
> > +	if s.TapeUsed == false {
> > +		w.WriteHeader(503)
> > +		fmt.Fprintln(w, items.ErrNoStore)
>
> Using items here is so random. These routes otherwise don't touch items.
> Why bring it in just for this? If you want consistency, there must be
> another way?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ndlib/bendo/pull/136/files/116b45889c785d5598950eba80d6730d33f98945#r76322302>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADNGTLkJURsNgmQrxOxMwWsWsmAab8L-ks5qjgCrgaJpZM4JsckT>
> .
>